atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java

```diff
@@ -204,6 +204,20 @@ private void writeStreamToFile(Transaction transaction, T id, StreamMetadata met
         }
     }
 
+    private void loadNBlocksToOutputStream(
+            Transaction tx,
+            T streamId,
+            long firstBlock,
+            long numBlocks,
+            OutputStream os) {
+        long position = 0;
+        for (long i = 0; i < numBlocks; i += 4) {
+            loadBlocksToOutputStream(tx, streamId, firstBlock + i, 4, os);
+            position += i;
```

Comment:
```suggestion
            position = i;
```

You want the last value where you left off. If there are 9 blocks, this would end up being 12, but you probably want it to be 8.